mlflow/utils/environment.py

```diff
@@ -82,3 +89,61 @@ def _get_additional_pip_dep(conda_env):
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
     pip_deps = _get_additional_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _get_pip_requirements(module_names):
+    module_to_packages = importlib_metadata.packages_distributions()
+    requirements = []
+    for module_name in module_names:
+        if module_name not in module_to_packages:
+            continue
+
+        for pkg in module_to_packages[module_name]:
+            req = "{}=={}".format(pkg, importlib_metadata.version(pkg))
+            if req in requirements:
+                continue
+
+            requirements.append(req)
+
+    return sorted(requirements)
+
+
+def _infer_pip_requirements(model_uri):
+    code_template = """
+import json
+import sys
+
+import mlflow
+
+def get_loaded_modules():
+   loaded_modules = set()
+   for mod_name in sys.modules.keys():
+       if not mod_name.startswith("_"):
+           loaded_modules.add(mod_name.split(".")[0])
+   return loaded_modules
+
+model = mlflow.pyfunc.load_model("{mode_uri}")
```

Comment:
```suggestion
model = mlflow.pyfunc.load_model("{model_uri}")
```

Reply:
This subprocess code may not have the correct tracking URI set, right? If that's the case, MLflow may not be able to download the model files from `model_uri`. Instead, I think we should call [mlflow.utils.artifact_utils._download_artifact_from_uri](https://github.com/mlflow/mlflow/blob/bc86f1dfdb6444746470b7b12b6582b1390eb1aa/mlflow/tracking/artifact_utils.py#L56) prior to creating the new model loading process.